optimizer_wrapper.rb

```diff
@@ -226,9 +226,14 @@ def self.solve(service_vrp, job = nil, block = nil)
         periodic = Interpreters::PeriodicVisits.new(vrp)
         vrp = periodic.expand(vrp, job, &block)
         optim_solution = vrp.configuration.preprocessing.heuristic_result if vrp.periodic_heuristic?
+        if vrp.configuration.resolution.solver && vrp.configuration.preprocessing.first_solution_strategy == ['periodic']
+          vrp.configuration.preprocessing.first_solution_strategy = ['global_cheapest_arc']
+        end
       end
 
-      if vrp.configuration.resolution.solver && !vrp.periodic_heuristic?
+      if vrp.configuration.resolution.solver &&
+         (vrp.configuration.preprocessing.partitions.empty? && vrp.configuration.preprocessing&.max_split_size.nil? ||
+           vrp.services.size < (vrp.configuration.preprocessing&.max_split_size || 0))
```

Comment:
Why do we need these extra conditions? Max split and partition logics are handled before `solve()`. 

Is it because you want to limit the size of the general problem that is optimized with or-tools directly as we talked? 

If that is the case what I expected was a more literal condition like this 

```suggestion
      if vrp.configuration.resolution.solver && (
        !vrp.periodic_heuristic? || vrp.services.size < XXX
      )
```

This way we are sure that we do not affect the behaviour of other options. Plus if we the user increase max_split for any reason this would lead to ortools being launched with too many services or if they do not set max_split we might launch an optim with 10K services (which would definitely lead to an OOM).

Since with the above modification I suggested we are sure that this check will be active after periodic_heuristic is already launched and we certainly have an initial solution, I suggest a value like 500 for `XXX`.